test.sh

```diff
@@ -71,7 +69,7 @@ function setup_dfp() {
 
   # CentOS needs to have setuptools updates to make pytest-cov work
   if [[ $OS != "fedora" ]]; then
-    $RUN $PIP install -U setuptools
+    $RUN $PIP install -U 'setuptools<45'
```

Comment:
You don't have to pin to a specific version here. As long as you

```bash
  $RUN $PIP install -U pip
  $RUN $PIP install -U setuptools
```

pip and setuptools will do the right thing

Reply:
because old pip cannot understand supported python version listed in packages. Upgraded pip will install setuptools=44 under py2 as expected

Reply:
@iciclespider I just had to handle this exact issue in [osbs-client](https://github.com/containerbuildsystem/osbs-client/pull/984/files) - as long as you `$RUN $PIP install -U pip`, pip will do the right thing and install[1] a correct, working setuptools for the OS/Python it's running under.

[1] I did `$RUN $PIP install -U setuptools` immediately after updating pip; YMMV

Reply:
Oh, I see what you are saying. Added the pip upgrade before the setuptools upgrade.